### PR TITLE
LPS-144201 Move usages of java-diff.jar and daisydiff.jar to a new module

### PR DESCRIPTION
**Pull request by guilhermedcamacho**

### CODE REVIEW

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
Hey @tinatian , related to things you mentioned [here](https://github.com/liferay-core-infra/liferay-portal/pull/104#issuecomment-1002890762):

1. I ran **ant build-lib-versions** and force pushed the changes.
2. I've checked and it's not easy to replace "com.googlecode.java-diff-utils" to use the new module "diff-util" in project-templates and source-formatter projects.
3. I already followed the rules pointed by the [document](https://liferay.atlassian.net/wiki/spaces/TEAMRM/pages/1662878423/How+to+add+a+new+app+to+the+release+bundle.), when this code is merged to upstream, I will have to open an LRQA ticket asking to create the file app.properties

I also rebased this PR. Should we run ci:test again for this pull? Thanks!

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test

**Comment by guilhermedcamacho** [comment]
Hey @vicnate5 , please help me scan test results. Thanks!

**Comment by guilhermedcamacho** [comment]
Fixed code conflicts with these files, but the version stayed the same 7.0.0
modules/apps/frontend-taglib/frontend-taglib/bnd.bnd
modules/apps/frontend-taglib/frontend-taglib/package.json

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by vicnate5** [comment]
@guilhermedcamacho No regressions found. I checked the default results from before the rebase, and then the relevant results after the rebase.

**Comment by tinatian** [comment]
@shuyangzhou This pull is good to go.

Tina

File: modules/apps/diff/diff-impl/build.gradle [hunk 1]
```diff
@@ -0,0 +1,12 @@
+dependencies {
+	compileInclude group: "com.liferay", name: "org.outerj.daisy.diff", version: "1.2.0-atlassian-hosted.LIFERAY-PATCHED-1"
+	compileInclude group: "net.sourceforge.nekohtml", name: "nekohtml", version: "1.9.22"
+	compileInclude group: "org.eclipse.core", name: "runtime", version: "20070801"
```
Comment (shuyangzhou) [inline modules/apps/diff/diff-impl/build.gradle:4]: @guilhermedcamacho shouldn't this ```compileInclude group: "org.eclipse.core", name: "runtime", version: "20070801"``` be removed from lib/portal? Any reason to include it here and still keep the lib/portal one?

Comment (guilhermedcamacho) [reply modules/apps/diff/diff-impl/build.gradle:4]: No reason to keep it, I've just removed and force pushed, thanks!

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by vicnate5** [comment]
@Hijky can you check these new results on Monday?

**Comment by guilhermedcamacho** [comment]
ci:test

**Comment by Hijky** [comment]
> @Hijky can you check these new results on Monday?

I will check these today

**Comment by Hijky** [comment]
None of the failed cases are relevant

**Comment by guilhermedcamacho** [comment]
Hey @shuyangzhou , I did the changes you asked and the tests passed , can you review again? Thanks!

**Comment by shuyangzhou** [comment]
See https://github.com/shuyangzhou/liferay-portal/pull/10516

**Comment by shuyangzhou** [comment]
ci:close

**Pull request closed without merging**